tests/aws/services/cloudformation/api/test_nested_stacks.py

```diff
@@ -291,3 +292,59 @@ def test_nested_stacks_conditions(deploy_cfn_template, s3_create_bucket, aws_cli
         StackName=stack.outputs["NestedStackArn"]
     )
     assert ":" not in nested_stack["Stacks"][0]["StackName"]
+
+
+@markers.aws.validated
+def test_deletion_of_failed_nested_stack(s3_create_bucket, aws_client, region_name):
+    """
+    This test confirms that after deleting a stack parent with a failed nested stack. The nested stack is also deleted
+    """
+
+    bucket_name = s3_create_bucket()
+    aws_client.s3.upload_file(
+        os.path.join(
+            os.path.dirname(__file__), "../../../templates/cfn_failed_nested_stack_child.yml"
+        ),
+        Bucket=bucket_name,
+        Key="child.yml",
+    )
+
+    stack_name = f"stack-{short_uid()}"
+    child_template_url = (
+        f"https://{bucket_name}.s3.{config.LOCALSTACK_HOST.host_and_port()}/child.yml"
+    )
+    if is_aws_cloud():
+        child_template_url = f"https://{bucket_name}.s3.{region_name}.amazonaws.com/child.yml"
+
+    aws_client.cloudformation.create_stack(
+        StackName=stack_name,
+        TemplateBody=load_file(
+            os.path.join(
+                os.path.dirname(__file__), "../../../templates/cfn_failed_nested_stack_parent.yml"
+            ),
+        ),
+        Parameters=[
+            {"ParameterKey": "BucketName", "ParameterValue": "Invalid!@#$"},
+            {"ParameterKey": "TemplateUri", "ParameterValue": child_template_url},
+        ],
+    )
+
+    def _status_is_complete():
+        status = aws_client.cloudformation.describe_stacks(StackName=stack_name)["Stacks"][0][
+            "StackStatus"
+        ]
+        return status in [
+            "ROLLBACK_COMPLETE",
+            "CREATE_FAILED",
+        ]  # LS doesn't implement the ROLLBACK yet
+
+    wait_until(_status_is_complete, strategy="static")
+
+    aws_client.cloudformation.delete_stack(StackName=stack_name)
+
+    def _stack_and_child_deleted():
+        stacks = aws_client.cloudformation.describe_stacks()["Stacks"]
+        stacks_with_name = [s for s in stacks if stack_name in s["StackName"]]
+        assert len(stacks_with_name) == 0
+
+    retry(_stack_and_child_deleted, sleep=2 if is_aws_cloud() else 1)
```

Comment:
Could you use a [waiter](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudformation/waiter/StackDeleteComplete.html) here? Also it might be clearer to `describe_stack(StackName=stack_name)` and catch the exception?

Reply:
Took a while but test is a lot better now. I'll merge once is green.